Add auth with MS Entra ID to docs (#2499)
chainguard-dev/edu: content/chainguard/chainguard-images/chainguard-registry/authenticating/index.md

See the [CircleCI documentation](https://circleci.com/docs/openid-connect-tokens/#format-of-the-openid-connect-id-token) to learn more about using OpenID Connect tokens in CircleCI jobs.

## Authenticating with Microsoft Entra ID OIDC Token

You can configure authentication with OIDC using Microsoft Entra ID (formerly Azure Active Directory).

Acquiring an OIDC ID token in Entra requires completing an OAuth 2.0/OIDC flow. Entra issues access tokens (for authorization) and ID tokens (for authentication) as separate but related JWTs. Access tokens grant API access, while ID tokens prove user identity.

If you use the implicit or hybrid flows, in Entra ID enable **ID tokens (used for implicit and hybrid flows)** under **Authentication → Implicit grant and hybrid flows** in for your application, and configure a redirect URI as described in [Enable ID tokens](https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols-oidc#enable-id-tokens).

> For authorization code flow (recommended), request the `openid` scope to receive an ID token; no portal checkbox is required. Then, [authenticate a user and request an ID token](https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-auth-code-flow#request-an-authorization-code).

Reviewer: ```suggestion
For the authorization code flow (recommended), request the `openid` scope to receive an ID token; you don't need to select any of the portal checkboxes. Then, [authenticate a user and request an ID token](https://learn.microsoft.com/en-us/entra/identity-platform/v2-oauth2-auth-code-flow#request-an-authorization-code).
```
